hello_world/src/hello-world.cpp

```diff
@@ -1,3 +1,24 @@
 #include "hello-world.hpp"
 
+#ifdef USING_CMAKEPUBLIC_
+
+std::string say_goodbye() {
+    int val = cmake_public::call_cmake_public();
+    std::string str = std::to_string(val);
+    return (std::string("Goodbye from CMake Public! ") + str);
+}
+
+#else
+
+std::string say_goodbye() {
+    return (std::string("Goodbye from CMake superbuild!"));
+}
+
+#endif
+
 std::string say_hello() { return std::string("Hello, CMake superbuild!"); }
+
+int main() {
```

Comment:
This is a library so there's no main function. If you want, for a unit test you can make an executable that links to the library.